cmd/lotus-chainwatch/processor/mpool.go

```diff
@@ -0,0 +1,103 @@
+package processor
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+func (p *Processor) subMpool(ctx context.Context) {
+	sub, err := p.node.MpoolSub(ctx)
+	if err != nil {
+		return
+	}
+
+	for {
+		var updates []api.MpoolUpdate
+
+		select {
+		case update := <-sub:
+			updates = append(updates, update)
+		case <-ctx.Done():
+			return
+		}
+
+	loop:
+		for {
+			time.Sleep(10 * time.Millisecond)
+			select {
+			case update := <-sub:
+				updates = append(updates, update)
+			default:
+				break loop
+			}
+		}
```

Comment:
This looks like you're trying to make the loop run on a timer? What's the motivation with these loops?

Reply:
This is copy-pasted from https://github.com/filecoin-project/lotus/blob/3f017688c7ae180d12d4ee28f732298299ee351f/cmd/lotus-chainwatch/mpool.go.